packages/libs/src/sdk/services/Solana/programs/BaseSolanaProgramClient.ts

```diff
@@ -90,14 +105,41 @@ export class BaseSolanaProgramClient {
       instructions,
       feePayer,
       recentBlockhash,
-      addressLookupTables = []
+      addressLookupTables = [],
+      priorityFee
     } = await parseParams('buildTransaction', BuildTransactionSchema)(params)
 
     if (!recentBlockhash) {
       const res = await this.connection.getLatestBlockhash()
       recentBlockhash = res.blockhash
     }
 
+    if (priorityFee !== undefined) {
+      if ('microLamports' in priorityFee) {
+        instructions.push(
+          ComputeBudgetProgram.setComputeUnitPrice({
+            microLamports: priorityFee.microLamports
+          })
+        )
+      } else {
+        const res = await this.connection.getRecentPrioritizationFees()
+        const orderedFees = res.map((r) => r.prioritizationFee).sort()
+        const percentile =
+          'percentile' in priorityFee
+            ? priorityFee.percentile
+            : priorityToPercentileMap[priorityFee.priority]
+        const microLamports =
+          orderedFees[Math.round((percentile / 100.0) * orderedFees.length)]
```

Comment:
am i crazy - if `percentile` is 100, would this cause array out of bounds? (100/100 = 1) * length = length.

Reply:
or i guess not out of bounds in js but we wouldn't get a value when we expected to?

Reply:
good call, will clamp